motionplan/tpspace/ptg.go

```diff
@@ -20,13 +21,14 @@ var flipPose = spatialmath.NewPoseFromOrientation(&spatialmath.OrientationVector
 
 // PTGSolver wraps a PTG with the ability to perform Inverse Kinematics.
 type PTGSolver interface {
-	// Solve will return the (alpha, dist) TP-space coordinates whose corresponding relative pose minimizes the given function
-	ik.InverseKinematics
+	referenceframe.Limited
```

Comment:
I think we want to keep this how it was no?

Reply:
On the fence here. We were writing a lot of boilerplate dealing with PTG solvers returning things via the channel, which is entirely unnecessary for PTGs specifically.

On the other hand, the grid sim solver especially *is* a solver.

I could go either way.

Reply:
In tpspace and in course correction, we're replacing:
```
solutionChan := make(chan *ik.Solution, 1)
err := ptg.Solve(context.Background(), solutionChan, seed, targetFunc, 0)
var solution *ik.Solution
select {
	case solution = <-solutionChan:
	default:
}
```
with
```
solution, err := ptg.Solve(context.Background(), seed, targetFunc)
```

Reply:
Yeah I could go either way on this, whatever you want to do is good with me